components/forwarder/publisher.go

```diff
@@ -0,0 +1,64 @@
+package forwarder
+
+import (
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+type PublisherConfig struct {
+	// ForwarderTopic is a topic which the forwarder is listening to. Publisher will send enveloped messages to this topic.
+	// Defaults to `forwarder_topic`.
+	ForwarderTopic string
+}
+
+func (c *PublisherConfig) setDefaults() {
+	if c.ForwarderTopic == "" {
+		c.ForwarderTopic = defaultForwarderTopic
+	}
+}
+
+func (c *PublisherConfig) Validate() error {
+	if c.ForwarderTopic == "" {
+		return errors.New("empty forwarder topic")
+	}
+
+	return nil
+}
+
+// PublisherDecorator changes `Publish` method behavior so it wraps a sent message in an envelope
+// and sends it to the forwarder topic provided in the config.
+type PublisherDecorator struct {
```

Comment:
I don't know why, but "Decorator" in this context is a bit misleading for me 🤔 maybe just forwarder.Publisher will be enough?

Reply:
Yeah, I agree, 328724a